workflows/autometa_flagged.sh

```diff
@@ -0,0 +1,438 @@
+    exit
+fi
+
+if [ -d "$marker_db" ]; then
+        echo "$marker_db found."
+else 
+        echo "$marker_db does not exist."
+        exit
+fi
+
+#Create output folder
+outdir="${work_dir}/${simple_name}_Autometa_Output"
+if [ ! -d $outdir ]
+then mkdir -p $outdir
+fi
+
+# Default Autometa Parameters - If you don't know what you're doing, leave these alone!!
+kmer_size=5
+norm_method="am_clr" # choices: am_clr, clr, ilr
+pca_dimensions=50 # NOTE: must be greater than $embed_dimensions
+embed_dimensions=2 # NOTE: must be less than $pca_dimensions
+embed_method="bhsne" # choices: bhsne, sksne, umap, densne, trimap
+cluster_method="hdbscan" # choices: hdbscan, dbscan
+completeness=20.0
+purity=95.0
+cov_stddev_limit=25.0
+gc_stddev_limit=5.0
+seed=42
+
+# Report autometa version
+echo
+autometa --version
+
+#Step 1: Filter assembly by length and retrieve contig lengths as well as GC content
+
+filtered_assembly="${outdir}/${simple_name}.filtered.fna"
+gc_content="${outdir}/${simple_name}.gc_content.tsv"
+
+autometa-length-filter \
+    --assembly $assembly \
+    --cutoff $length_cutoff \
+    --output-fasta $filtered_assembly \
+    --output-gc-content $gc_content
+
+
+#Step 2: Determine coverages from assembly read alignments
+coverages="${outdir}/${simple_name}.coverages.tsv"
+
+if [[ "$coverage" == "spades" ]]
+then
+    echo
+    echo "Coverage to be taken from Spades headers"
+    autometa-coverage --assembly $filtered_assembly --from-spades --out $coverages
+else
+    if [[ "$coverage" == "paired" ]]
+    then
+        echo
+        echo "Coverage to be calculated from reads"
+        if [ -z $fwd_reads ]; then
+            echo "Forward reads were not provided. These are required for coverage calculations"
+            exit 1
+        fi
+
+        if [ -f "$fwd_reads" ]; then
+            echo "$fwd_reads found."
+        else 
+            echo "$fwd_reads does not exist."
+            exit
+        fi
+
+        if [ -z $rev_reads ]; then
+            echo "Reverse reads were not provided. These are required for coverage calculations"
+            exit 1
+        fi
+
+        if [ -f "$rev_reads" ]; then
+            echo "$rev_reads found."
+        else 
+            echo "$rev_reads does not exist."
+            exit
+        fi
+
+        autometa-coverage --assembly $filtered_assembly --fwd-reads $fwd_reads --rev-reads $rev_reads --out $coverages --cpus $cpus
+
+    else
+        echo
+        echo "You have provided an invalid option. The options for coverage are 'spades' or 'paired'. Please fix">&2; exit 1
+    fi
+fi
+
+# Step 3: Annotate and filter markers
+
+orfs="${outdir}/${simple_name}.orfs.faa"
+orfs_nuc="${outdir}/${simple_name}.orfs.fna"
+
+#Get ORFs
+autometa-orfs \
+    --assembly $filtered_assembly \
+    --output-nucls $orfs_nuc \
+    --output-prots $orfs \
+    --cpus $cpus
+
+if [ $? -eq 0 ]; then
+   echo "ORFs successfully generated"
+else
+   echo "ORF generation failed"
+fi
```

Comment:
This check is performed automagically with the `set -e` prepended to the workflow

```suggestion
```